Zend/zend_execute.c

```diff
@@ -3161,6 +3165,9 @@ static zend_always_inline void zend_fetch_property_address(zval *result, zval *c
 						ZEND_ASSERT(type == BP_VAR_W || type == BP_VAR_RW || type == BP_VAR_UNSET);
 						if (Z_TYPE_P(ptr) == IS_OBJECT) {
 							ZVAL_COPY(result, ptr);
+						} else if (Z_PROP_FLAG_P(ptr) & IS_PROP_REINITABLE) {
+							Z_PROP_FLAG_P(ptr) &= ~IS_PROP_REINITABLE;
+							ZVAL_COPY(result, ptr);
```

Comment:
I think this copy is incorrect.

```
--TEST--
__clone() can indirectly modify unlocked readonly properties
--FILE--
<?php

class Foo {
    public function __construct(
        public readonly array $bar
    ) {}

    public function __clone()
    {
        $this->bar['bar'] = 'bar';
    }
}

$foo = new Foo([]);
// First call fills the cache slot
var_dump(clone $foo);
var_dump(clone $foo);

?>
--EXPECT--
object(Foo)#2 (1) {
  ["bar"]=>
  array(1) {
    ["bar"]=>
    string(3) "bar"
  }
}
object(Foo)#2 (1) {
  ["bar"]=>
  array(0) {
  }
}
```

The second dump is wrong.

Reply:
Wow, thanks so much!